### PR TITLE
AP-2722 Remove ApplicationProceedingType from UsedDelegatedFunctionsController

### PR DESCRIPTION
**Pull request by stephenrichards**
## Remove ApplicationProceedingType from UsedDelegatedFunctionsController

[Link to story](https://dsdmoj.atlassian.net/browse/AP-2722)

The `UsedDelegatedFunctionsController` was set up to update both `Proceeding` records and `ApplicationProceedingType` records with DF date.  This PR removes the updates to the `ApplicationProceedingType`

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Comment by kmahern** [comment]
@stephenrichards I had a question, https://dsdmoj.atlassian.net/browse/AP-2722 talks about removing the dependencies on proceeding_type in the remove_delegated_scope_limitations, add_delegated_scope_limitations and update_delegated_scope_limitations methods in UsedDelegatedFunctionsController. I know when I looked at this controller previously I thought we couldn't yet do that (because proceeding_type is required by AddAssignedScopeLimitationService which is called in multiple places) - do you think that is still the case?

**Comment by stephenrichards** [comment]
Hi @kmahern - I'll look at this again, but I think that now the scope limitations data is held on the `Proceeding` record, and all the Check your answer pages, merits report and CCMS payload all use the `Proceeding` model for their data source, we don't really care any more what is on the application proceeding types scope limtiations records.  Having said that, I thought I'd removed a call to one of the services that takes care of all that, but I don't see that in the changes on this PR, so am going to take another close look.

Thanks for the heads-up

File: spec/forms/legal_aid_applications/used_multiple_delegated_functions_form_spec.rb [hunk 1]
```diff
@@ -50,8 +44,6 @@
         end
 
         it 'updates the application types with no reported on date' do
```
Comment (colinbruce) [inline spec/forms/legal_aid_applications/used_multiple_delegated_functions_form_spec.rb:46]: Should this description be changed to `updates the proccedings with no reported on date`?

**Review by colinbruce — APPROVED** [review]
One minor comment about a test description but, otherwise, LGTM 👍

**Pull request merged**